### PR TITLE
Add ISO8583-connector-configuration.md

### PR DESCRIPTION
**Pull request by KasunAratthanage**

### CODE REVIEW

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# ISO8583 Connector Configuration
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Rename to "Setting up ISO8583" or something similar.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+ISO8583 is an international standard for financial transaction card originated interchange messaging. It is the International Organization for Standardization standard for systems that exchange electronic transactions initiated by cardholders using payment cards.
+
+Typically, whenever we use a credit card or debit card or an ATM card the data travels from one system to another system. A card based trantraction typically needs to travel between number of systems.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]:
```suggestion
Typically, whenever we use a credit card, debit card, or ATM card, the data travels from one system to another system. A card-based transaction typically needs to travel between a number of systems.
```

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ISO8583 is a financial transaction card originated in touch message message specifications adopted by a wide segment of payment industry which include acquiring and issuers.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]:
```suggestion
ISO8583 is a financial transaction card originated in touch message specifications adopted by a wide segment of the payment industry, which includes acquiring and issuers.
```

What do you mean by "acquiring" here?

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+Before you start configuring the ISO8583 connector you also need WSO2 MI and we refer to that location as <PRODUCT_HOME>.
+
+To configure the ISO8583 connector, copy the following client libraries from the given locations to the `<PRODUCT_HOME>/repository/components/lib` directory.
+
+* [jpos-1.9.4.jar](http://mvnrepository.com/artifact/org.jpos/jpos/1.9.4)  
+* [jdom-1.1.3.jar](http://mvnrepository.com/artifact/org.jdom/jdom/1.1.3) 
+* [commons-cli-1.3.1.jar](http://mvnrepository.com/artifact/commons-cli/commons-cli/1.3.1) 
+
+>NOTE : For testing purposes, you need to have a test server (basically a java socket connection that listens on port 5010) to handle ISO8583 requests that come from the connector. You also need to generate responses by changing the relevant response fields, and then send the responses back to the connector. You can test the connector with the sample Java server program that is provided in the following [git location](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer)
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]:
```suggestion
> **Note** : For testing purposes, you need to have a test server (basically a Java socket connection that listens on port 5010) to handle ISO8583 requests that come from the connector. You also need to generate responses by changing the relevant response fields, and then send the responses back to the connector. You can test the connector with the sample Java server program that is provided in the following [git location](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer).
```

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+Now that you have connected to test server, For more information, see
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]:
```suggestion
Now you have connected to the test server. For more information, see
```

Merge the lines below with this one.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# ISO8583 Connector Configuration
+
+ISO8583 is an international standard for financial transaction card originated interchange messaging. It is the International Organization for Standardization standard for systems that exchange electronic transactions initiated by cardholders using payment cards.
+
+Typically, whenever we use a credit card or debit card or an ATM card the data travels from one system to another system. A card based trantraction typically needs to travel between number of systems.
+ISO8583 is a financial transaction card originated in touch message message specifications adopted by a wide segment of payment industry which include acquiring and issuers.
+
+## Setting up the environment
+
+Before you start configuring the ISO8583 connector you also need WSO2 MI and we refer to that location as <PRODUCT_HOME>.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]:
```suggestion
Before you start configuring the ISO8583 connector, you also need WSO2 MI, and we refer to that location as <PRODUCT_HOME>.
```

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 1 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Done.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 2 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Done.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 3 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Changed as shown bellow.
"ISO 8593 is Financial Transaction card originated, Interchange message specifications adopted by wide segment of Payment Industry (including Acquirers and Issuers)."

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 4 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Updated.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 5 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Done.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 6 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Done.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+If you use the [sample Java server program](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer) to send an ISO8583 request with a header value from the connector, you need to update the iso87ascii.xml file with the relevant headerLength information.
+
+The ISO8583 connector uses the jpos library, which is a third party library that provides a high-performance bridge between card messages generated at point of sale terminals, ATMs, and internal systems across the entire financial messaging network. The jposdef.xml file has the field definitions of standard ISO8583 messages. According to the field definitions, each ISO8583 message in XML format coming from the REST client is packed and sent to the test server. Therefore, you need to create a file called jposdef.xml (with the contents given here) in the <EI_HOME> directory.
+ 
+Now you have connected to the test server. For more information, see
+
+[ISO8583 Connector Example](ISO8583-connector-example.md).
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Please merge these two lines.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+* [jpos-1.9.4.jar](http://mvnrepository.com/artifact/org.jpos/jpos/1.9.4)  
+* [jdom-1.1.3.jar](http://mvnrepository.com/artifact/org.jdom/jdom/1.1.3) 
+* [commons-cli-1.3.1.jar](http://mvnrepository.com/artifact/commons-cli/commons-cli/1.3.1) 
+
+## Configure the test server
+
+For testing purposes, you need to have a test server (basically a Java socket connection that listens on port 5010) to handle ISO8583 requests that come from the connector. You also need to generate responses by changing the relevant response fields, and then send the responses back to the connector. You can test the connector with the sample Java server program that is provided in the following [git location](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer). 
+
+You can include required header information within the header tag in . It supports 2-byte or 4-byte headers. To include header information, you need to convert the 2-byte or 4-byte header into a string using base64 encoding, and then specify the string value within the header tag.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]:
```suggestion
You can include required header information within the header tag. It supports 2-byte or 4-byte headers. To include header information, you need to convert the 2-byte or 4-byte header into a string using base64 encoding, and then specify the string value within the header tag.
```

Is there some example we can add here or link that we can point to regarding this? It sounds complicated so its better if we can give some more instruction.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 7 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Done.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
+# Setting up ISO8583 Connector 
+
+ISO8583 is an international standard for financial transaction card originated interchange messaging. It is the International Organization for Standardization standard for systems that exchange electronic transactions initiated by cardholders using payment cards.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: "ISO8583 is an international standard for financial transaction card originated interchange messaging." Can we simplify this? I see some similar text in wikipedia and it's a bit too complex. Can we simplify this definition?

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+Typically, whenever we use a credit card, debit card, or ATM card, the data travels from one system to another system. A card-based transaction typically needs to travel between a number of systems.
+ISO 8593 is Financial Transaction card originated, Interchange message specifications adopted by wide segment of Payment Industry (including Acquirers and Issuers).
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: The "Financial Transaction card originated," part doesn't make much sense here. Can we reword that? 

Also change to ISO8583.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+You can include required header information within the header tag in . It supports 2-byte or 4-byte headers. To include header information, you need to convert the 2-byte or 4-byte header into a string using base64 encoding, and then specify the string value within the header tag.
+
+If you use the [sample Java server program](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer) to send an ISO8583 request with a header value from the connector, you need to update the iso87ascii.xml file with the relevant headerLength information.
+
+The ISO8583 connector uses the jpos library, which is a third party library that provides a high-performance bridge between card messages generated at point of sale terminals, ATMs, and internal systems across the entire financial messaging network. The jposdef.xml file has the field definitions of standard ISO8583 messages. According to the field definitions, each ISO8583 message in XML format coming from the REST client is packed and sent to the test server. Therefore, you need to create a file called jposdef.xml (with the contents given here) in the <EI_HOME> directory.
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: The "with the contents given here" part is confusing. What contents exactly?

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 8 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Simply added reference document for this part.(https://en.wikipedia.org/wiki/ISO_8583).

This section (Configure the test server) not mandatory configuration for setting up the connector. If customer need to set up this connector with their scenarios, they should have their own ISO8583 back-ends. This sample Java server program just only simulates the ISO8583 mock back end behaviors.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 9 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Updated.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 10 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Updated.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md (hunk 11 above)
Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Added link of the repository.

File: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+* [jpos-1.9.4.jar](http://mvnrepository.com/artifact/org.jpos/jpos/1.9.4)  
+* [jdom-1.1.3.jar](http://mvnrepository.com/artifact/org.jdom/jdom/1.1.3) 
+* [commons-cli-1.3.1.jar](http://mvnrepository.com/artifact/commons-cli/commons-cli/1.3.1) 
+
+## Configure the test server
+
+For testing purposes, you need to have a test server (basically a Java socket connection that listens on port 5010) to handle ISO8583 requests that come from the connector. You also need to generate responses by changing the relevant response fields, and then send the responses back to the connector. You can test the connector with the sample Java server program that is provided in the following [git location](https://github.com/wso2-docs/CONNECTORS/tree/master/ISO8583/ISO8583TestServer). 
+
+You can include required header information within the header tag. It supports 2-byte or 4-byte headers. To include header information, you need to convert the 2-byte or 4-byte header into a string using base64 encoding, and then specify the string value within the header tag.For more information on the ISO8583 standard, see [ISO8583 Documentation](https://en.wikipedia.org/wiki/ISO_8583).
```
Comment (samgnaniah) [inline en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]:
```suggestion
You can include required header information within the header tag. It supports 2-byte or 4-byte headers. To include header information, you need to convert the 2-byte or 4-byte header into a string using base64 encoding, and then specify the string value within the header tag. For more information on the ISO8583 standard, see [ISO8583 Documentation](https://en.wikipedia.org/wiki/ISO_8583).
```
Wikipedia is not so reliable. Do we have a more reliable link?

Comment (KasunAratthanage) [reply en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-configuration.md]: Updated.

**Pull request merged**